cli/src/main.rs

```diff
@@ -133,43 +133,29 @@ fn main() {
             "Could not save minified code"
         );
     } else {
-        let (mut tx, rx) = spmc::channel::<PathBuf>();
-        let mut handles = Vec::new();
-        for _ in 0..num_cpus::get() {
-            let rx = rx.clone();
-            let cfg = cfg.clone();
-            handles.push(thread::spawn(move || {
-                let input = rx.recv().unwrap();
-                let input_name = input.to_string_lossy().into_owned();
-
-                let mut src_file =
-                    io_expect!(input_name, File::open(&input), "Could not open source file");
-                let mut src_code = Vec::<u8>::new();
-                io_expect!(
-                    input_name,
-                    src_file.read_to_end(&mut src_code),
-                    "Could not load source code"
-                );
-                let out_code = minify(&src_code, &cfg);
-                let mut out_file = io_expect!(
-                    input_name,
-                    File::create(&input),
-                    "Could not open output file"
-                );
-                io_expect!(
-                    input_name,
-                    out_file.write_all(&out_code),
-                    "Could not save minified code"
-                );
-            }));
-        }
-
-        for i in args.inputs {
-            tx.send(i).unwrap();
-        }
+        args.inputs.par_iter().for_each(|input| {
+            let input_name = input.to_string_lossy().into_owned();
 
-        for handle in handles {
-            handle.join().unwrap();
-        }
+            let mut src_file =
+                io_expect!(input_name, File::open(&input), "Could not open source file");
+            let mut src_code = Vec::<u8>::new();
+            io_expect!(
+                input_name,
+                src_file.read_to_end(&mut src_code),
+                "Could not load source code"
+            );
+            let out_code = minify(&src_code, &cfg);
+            let mut out_file = io_expect!(
+                input_name,
+                File::create(&input),
+                "Could not open output file"
+            );
+            io_expect!(
+                input_name,
+                out_file.write_all(&out_code),
+                "Could not save minified code"
+            );
+            println!("finish minifying {}", input_name);
```

Comment:
BTW the `println!` here will lock stdout, this may hurt parallelism if the previous minifying steps don't take much time. But I guess it is OK here and we can come back and optimize it in the future if it is a bottleneck.

Reply:
Yeah it seems fine, we can revisit in the future if need be. I think it should spend most of the time minifying and locking stdout should be relatively quicker.